The python output resolves that issue, next is `darwin.ICU` that doesn't build.

```
1 warning generated.
   c++	 ...  japancal.cpp
   c++	 ...  gregoimp.cpp
   c++	 ...  hebrwcal.cpp
   c++	 ...  indiancal.cpp
   c++	 ...  chnsecal.cpp
   c++	 ...  cecal.cpp
   c++	 ...  coptccal.cpp
   c++	 ...  nfsubs.cpp
   c++	 ...  dangical.cpp
   c++	 ...  ethpccal.cpp
   c++	 ...  coleitr.cpp
   c++	 ...  coll.cpp
   c++	 ...  sortkey.cpp
   c++	 ...  bocsu.cpp
   c++	 ...  ucoleitr.cpp
   c++	 ...  ucol.cpp
ucoleitr.cpp:323:28: error: qualified reference to 'UCollationPCE' is a constructor name rather than a type in this context
    return (UCollationPCE::UCollationPCE(elems)).nextProcessed(ixLow, ixHigh, status);
                           ^
ucoleitr.cpp:387:28: error: qualified reference to 'UCollationPCE' is a constructor name rather than a type in this context
    return (UCollationPCE::UCollationPCE(elems)).previousProcessed(ixLow, ixHigh, status);
                           ^
2 errors generated.
*** Failed compilation command follows: ----------------------------------------------------------
c++ -DU_ATTRIBUTE_DEPRECATED= -DU_I18N_IMPLEMENTATION -DSTD_INSPIRED -DMAC_OS_X_VERSION_MIN_REQUIRED=1070 -DU_HAVE_ATOMIC=1 -DU_DISABLE_RENAMING=1 -I. -I../common -DU_SHOW_CPLUSPLUS_API=1 -DU_SHOW_INTERNAL_API=1 -DU_TIMEZONE=timezone -DICU_DATA_DIR="/usr/share/icu/" -g -Os -Wglobal-constructors -fno-exceptions -fvisibility=hidden -fvisibility-inlines-hidden -fno-common -c -dynamic -MMD -MT ucoleitr.d ucoleitr.o ucoleitr.ao -o ucoleitr.o ucoleitr.cpp
```

--- a/pkgs/development/compilers/zig/default.nix
+++ b/pkgs/development/compilers/zig/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchFromGitHub, cmake, llvmPackages_5, llvm_5 }:
+{ stdenv, fetchFromGitHub, cmake, llvmPackages }:
 
 stdenv.mkDerivation rec {
   version = "0.1.1";
@@ -11,7 +11,7 @@ stdenv.mkDerivation rec {
     sha256 = "01yqjyi25f99bfmxxwyh45k7j84z0zg7n9jl8gg0draf96mzdh06";
   };
 
-  buildInputs = [ cmake llvmPackages_5.clang-unwrapped llvm_5 ];
+  buildInputs = [ cmake llvmPackages.clang-unwrapped llvmPackages.llvm ];
 
   cmakeFlags = [
     "-DZIG_LIBC_INCLUDE_DIR=${stdenv.cc.libc_dev}/include"

--- a/pkgs/development/haskell-modules/configuration-nix.nix
+++ b/pkgs/development/haskell-modules/configuration-nix.nix
@@ -258,7 +258,7 @@ self: super: builtins.intersectAttrs super {
       }
     );
 
-  llvm-hs = super.llvm-hs.override { llvm-config = pkgs.llvm_5; };
+  llvm-hs = super.llvm-hs.override { llvm-config = pkgs.llvm; };
 
   # Needs help finding LLVM.
   spaceprobe = addBuildTool super.spaceprobe self.llvmPackages.llvm;

--- a/pkgs/os-specific/darwin/apple-source-releases/ICU/clang-5.patch
+++ b/pkgs/os-specific/darwin/apple-source-releases/ICU/clang-5.patch
@@ -1,0 +1,22 @@
+diff --git a/icuSources/i18n/ucoleitr.cpp b/icuSources/i18n/ucoleitr.cpp
+index ecc94c9..936452f 100644
+--- a/icuSources/i18n/ucoleitr.cpp
++++ b/icuSources/i18n/ucoleitr.cpp
+@@ -320,7 +320,7 @@ ucol_nextProcessed(UCollationElements *elems,
+                    int32_t            *ixHigh,
+                    UErrorCode         *status)
+ {
+-    return (UCollationPCE::UCollationPCE(elems)).nextProcessed(ixLow, ixHigh, status);
++    return (UCollationPCE(elems)).nextProcessed(ixLow, ixHigh, status);
+ }
+ 
+ 
+@@ -384,7 +384,7 @@ ucol_previousProcessed(UCollationElements *elems,
+                    int32_t            *ixHigh,
+                    UErrorCode         *status)
+ {
+-    return (UCollationPCE::UCollationPCE(elems)).previousProcessed(ixLow, ixHigh, status);
++    return (UCollationPCE(elems)).previousProcessed(ixLow, ixHigh, status);
+ }
+ 
+ U_NAMESPACE_BEGIN

--- a/pkgs/os-specific/darwin/apple-source-releases/ICU/default.nix
+++ b/pkgs/os-specific/darwin/apple-source-releases/ICU/default.nix
@@ -3,6 +3,8 @@
 appleDerivation {
   nativeBuildInputs = [ cctools ];
 
+  patches = [ ./clang-5.patch ];
+
   postPatch = ''
     substituteInPlace makefile \
       --replace /usr/bin/ "" \

--- a/pkgs/os-specific/linux/bcc/default.nix
+++ b/pkgs/os-specific/linux/bcc/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchFromGitHub, fetchpatch, makeWrapper, cmake, llvmPackages_5, kernel
+{ stdenv, fetchFromGitHub, fetchpatch, makeWrapper, cmake, llvmPackages, kernel
 , flex, bison, elfutils, python, pythonPackages, luajit, netperf, iperf, libelf }:
 
 stdenv.mkDerivation rec {
@@ -13,7 +13,7 @@ stdenv.mkDerivation rec {
   };
 
   buildInputs = [
-    llvmPackages_5.llvm llvmPackages_5.clang-unwrapped kernel
+    llvmPackages.llvm llvmPackages.clang-unwrapped kernel
     elfutils python pythonPackages.netaddr luajit netperf iperf
   ];
 

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -6251,7 +6251,7 @@ with pkgs;
   llvm_35 = llvmPackages_35.llvm;
   llvm_34 = llvmPackages_34.llvm;
 
-  llvmPackages = recurseIntoAttrs llvmPackages_4;
+  llvmPackages = recurseIntoAttrs llvmPackages_5;
 
   llvmPackagesSelf = llvmPackages_34.override {
     stdenv = libcxxStdenv;
@@ -6277,17 +6277,17 @@ with pkgs;
     inherit (stdenvAdapters) overrideCC;
   };
 
-  llvmPackages_4 = callPackage ../development/compilers/llvm/4 ({
+  llvmPackages_4 = callPackage ../development/compilers/llvm/4 {
+    inherit (stdenvAdapters) overrideCC;
+  };
+
+  llvmPackages_5 = callPackage ../development/compilers/llvm/5 ({
     inherit (stdenvAdapters) overrideCC;
   } // stdenv.lib.optionalAttrs stdenv.isDarwin {
     cmake = cmake.override { isBootstrap = true; };
     libxml2 = libxml2.override { pythonSupport = false; };
     python2 = callPackage ../development/interpreters/python/cpython/2.7/boot.nix { inherit (darwin) CF configd; };
   });
-
-  llvmPackages_5 = callPackage ../development/compilers/llvm/5 {
-    inherit (stdenvAdapters) overrideCC;
-  };
 
   manticore = callPackage ../development/compilers/manticore { };
 
@@ -7944,7 +7944,6 @@ with pkgs;
 
   ycmd = callPackage ../development/tools/misc/ycmd {
     inherit (darwin.apple_sdk.frameworks) Cocoa;
-    llvmPackages = llvmPackages_5;
     python = python2;
   };
 
@@ -10100,8 +10099,6 @@ with pkgs;
     # makes it slower, but during runtime we link against just mesa_drivers
     # through /run/opengl-driver*, which is overriden according to config.grsecurity
     # grsecEnabled = true; # no more support in nixpkgs ATM
-
-    llvmPackages = llvmPackages_5;
   });
 
   mesa_glu =  mesaDarwinOr (callPackage ../development/libraries/mesa-glu { });

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -5611,9 +5611,7 @@ with pkgs;
     '';
   });
 
-  crystal = callPackage ../development/compilers/crystal {
-    llvm = llvm_5;
-  };
+  crystal = callPackage ../development/compilers/crystal { };
 
   devpi-client = callPackage ../development/tools/devpi-client {};
 

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -9935,7 +9935,7 @@ in {
 
   locustio = callPackage ../development/python-modules/locustio { };
 
-  llvmlite = callPackage ../development/python-modules/llvmlite {llvm=pkgs.llvm_5;};
+  llvmlite = callPackage ../development/python-modules/llvmlite { llvm = pkgs.llvm; };
 
   lockfile = buildPythonPackage rec {
     pname = "lockfile";

